skimage/io/collection.py

```diff
@@ -147,24 +160,28 @@ def imread_convert(f):
     >>> coll[0].shape
     (200, 200)
 
-    >>> ic = io.ImageCollection('/tmp/work/*.png:/tmp/other/*.jpg')
+    >>> ic = io.ImageCollection(['/tmp/work/*.png', '/tmp/other/*.jpg'])
```

Comment:
IMO, this should be the only supported way to setup multiple patterns in Scikit-Image 0.18.

To many options makes maintaining this kind of code super complicated.

Reply:
To do this, you would have to add a deprecation warning. In that case, I see this PR more as a "simplification" of the API, since lists of strings are much more intuitive for python users than a strange seperator. I would approve ;)

Reply:
As mentioned in the comments, let's just soft-deprecate the old behaviour. This means removing the documentation for the old syntax, including the second paragraph above. @stefanv do you agree with this approach?